libpromises/evalfunction.c

```diff
@@ -6637,33 +6598,62 @@ static FnCallResult FnCallMakerule(EvalContext *ctx, ARG_UNUSED const Policy *po
     {
         if (!S_ISREG(statbuf.st_mode))
         {
-            Log(LOG_LEVEL_WARNING, "Function 'makerule' target-file '%s' exists and is not a plain file", target);
+            Log(LOG_LEVEL_WARNING, "Function '%s' target-file '%s' exists and is not a plain file", fp->name, target);
             // Not a probe's responsibility to fix - but have this for debugging
         }
 
         target_time = statbuf.st_mtime;
     }
 
-    // For each file in sources, check they exist and are older than target
-
-    for (const Rlist *rp = list; rp != NULL; rp = rp->next)
+    // Check if the file name is explicit
+    if (lstat(name_str, &statbuf) != -1)
```

Comment:
> I'm not even sure what this function should do if the value is not scalar. Maybe just bail out?

Eh, no, that's the case below, obviously. So the only extra thing needed here is to check for scalar type (scalarity?) before the `lstat()`.